util/dvsim/NcLauncher.py

```diff
@@ -0,0 +1,225 @@
+        postcmd = (
+            f'{odir}/nc_post_cmd.sh >post.log; '
+            f'cat post.log >>{log_file}'
+        )
+        cmd_array = ['nc', 'run',
+                     '-e', 'SNAPSHOT',
+                     '-nodb', '-forcelog',
+                     '-l', log_file,
+                     '-postcmd', postcmd,
+                     '-set', job_name]
+        cmd_array.extend(interactive_flags)
+        self.create_run_sh(odir, cmd)
+        # TODO: This needs to be moved to a HJson.
+        if (exetool == 'xcelium'):
+            cmd_array.extend(['-r', 'License:Xcelium_Single_Core/1'])
+        if (exetool == 'vcs'):
+            cmd_array.extend(['-r', 'License:VCSRuntime_Net/1'])
+        cmd_array.extend(['--', f'{odir}/run.sh'])
+        return cmd_array
+
+    def _do_launch(self):
+        # Compute the environment for the subprocess by overriding environment
+        # variables of this process with matching ones from self.deploy.exports
+        exports = os.environ.copy()
+        if self.deploy.exports:
+            exports.update(self.deploy.exports)
+
+        # Clear the magic MAKEFLAGS variable from exports if necessary. This
+        # variable is used by recursive Make calls to pass variables from one
+        # level to the next. Here, self.cmd is a call to Make but it's
+        # logically a top-level invocation: we don't want to pollute the flow's
+        # Makefile with Make variables from any wrapper that called dvsim.
+        if 'MAKEFLAGS' in exports:
+            del exports['MAKEFLAGS']
+
+        self._dump_env_vars(exports)
+
+        if self.deploy.sim_cfg.interactive:
+            # Interactive: Set RUN_INTERACTIVE to 1
+            exports['RUN_INTERACTIVE'] = '1'
+            interactive_flags = ['-I']
+            cmd_arr = self.get_submit_cmd(self.deploy.sim_cfg.tool,
+                                          self.deploy.get_log_path(),
+                                          self.deploy.full_name,
+                                          self.deploy.cmd,
+                                          self.deploy.odir,
+                                          interactive_flags)
+            f = open(self.deploy.get_log_path(),
+                     'w',
+                     encoding='UTF-8',
+                     errors='surrogateescape')
+            f.write('[Executing]:\n{}\n\n'.format(self.deploy.cmd))
+            f.flush()
+            self.process = subprocess.Popen(cmd_arr,
+                                            stdin=None,
+                                            stdout=None,
+                                            stderr=subprocess.STDOUT,
+                                            # string mode
+                                            universal_newlines=True,
+                                            env=exports,
+                                            cwd=self.deploy.odir)
+
+            # Wait until the process exit
+            self.process.wait()
```

Comment:
For the blocking version that we're doing here, do we really need Popen? Isn't it equivalent to do this?
```python3
            subprocess.run(cmd_arr,
                           stdin=None,
                           stdout=None,
                           stderr=subprocess.STDOUT,
                           universal_newlines=True,
                           env=exports,
                           cwd=self.deploy.odir)
```

That way, we don't need to talk about `self.process` at all for interactive mode.

Reply:
I tried this, but it did not work. The reason is the poll function seems to get called even for interactive job and it has code that uses self.process variable. can i leave this as it is or should i rework the poll function for interactive job ?

Reply:
Hmm, the code is a bit ugly here. I think the call to `poll` comes from `Scheduler._poll`? Maybe the best approach is to set `self.status` after the `run` call has finished? Then `NcLauncher.poll` can do the same thing as `LsfLauncher.poll` and return it immediately.